hive/replays/recurrent_replay.py

```diff
@@ -0,0 +1,262 @@
+import os
+import pickle
+
+import numpy as np
+from hive.replays.circular_replay import CircularReplayBuffer
+
+
+class RecurrentReplayBuffer(CircularReplayBuffer):
+    """
+    First implementation of recurrent buffer without storing hidden states
+    """
+
+    def __init__(
+        self,
+        capacity: int = 10000,
+        max_seq_len: int = 1,
+        n_step: int = 1,
+        gamma: float = 0.99,
+        observation_shape=(),
+        observation_dtype=np.uint8,
+        action_shape=(),
+        action_dtype=np.int8,
+        reward_shape=(),
+        reward_dtype=np.float32,
+        extra_storage_types=None,
+        num_players_sharing_buffer: int = None,
+    ):
+        """Constructor for RecurrentReplayBuffer.
+
+        Args:
+            capacity (int): Total number of observations that can be stored in the
+                buffer. Note, this is not the same as the number of transitions that
+                can be stored in the buffer.
+            max_seq_len (int): The number of consecutive transitions in a sequence.
```

Comment:
Please clarify the description of this parameter.

Reply:
done